server/lib/validation.js

```diff
@@ -15,6 +15,7 @@ const PATTERNS = {
   BROKER: /^[0-9a-z-]+$/,
   CLIENT_ID: /^[0-9a-f]{16}/,
   CONTEXT: /^[0-9a-z_-]+$/,
+  DOMAIN: /^[\w.-]+$/,
```

Comment:
This would match `_` characters as well, which I don't believe are valid domains?

Reply:
I did not know that, thanks.